pkg/db/jsonType.go

```diff
@@ -0,0 +1,60 @@
+package db
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	"github.com/justtrackio/gosoline/pkg/mdl"
+)
+
+var (
+	_                        driver.Valuer = JSONType[struct{}]{}
+	_                        sql.Scanner   = &JSONType[struct{}]{}
+	ErrJSONInvalidDriverType               = errors.New("invalid incoming driver type for json type")
+)
+
+type (
+	JSONType[T any] struct {
+		val        T
+		asJsonNull bool
+	}
+)
+
+func NewJSONType[T any](val T) JSONType[T] {
+	return JSONType[T]{
+		val: val,
+	}
+}
+
+func (t *JSONType[T]) AsJSONNull() JSONType[T] {
```

Comment:
Do we need this to be a pointer receiver? Right now we could do
```go
jt.AsJSONNull()
```
as well as
```go
jt2 := jt1.AsJSONNull()
```
However, in this second case I could expect `jt1` to stay as it is as I am creating a new object with my change. Which would be wrong as `jt1` as well as `jt2` have the null flag set.
Finally, I think this won't work, right?
```go
jt := NewJSONType[t](getT()).AsJSONNull()
```

Finally, I guess a comment would be nice what this method changes (it turns a database null into the string `null` which is a JSON null, right?)

Reply:
Yes that was my idea that both of the first 2 code snippets are valid.
If that is unintuitive I guess I would remove the pointer receiver and have it just as example 2.

Regarding snippet 3:
You cant do that but you would need to wrap the NewJSONType Call in a function like mdl.Box.